app/controllers/api/audio_files_controller.rb

```diff
@@ -7,17 +7,17 @@ class Api::AudioFilesController < Api::BaseController
   filter_resources_by :audio_version_id
 
   # if the audio version is not set, and story is, set default av
-  def resource
+  def create_resource
     super.tap do |af|
-      if story && af.try(:new_record?)
+      if story
         af.audio_version_id ||= story.default_audio_version.id
         af.account_id ||= story.account_id
```

Comment:
We talked abut about this as a validation - to make sure these are in sync.
We could also have the model do a `#before_validation` method to make sure the `audio_file.account_id` is set to the `audio_version.account_id` when the `audio_file.audio_version_id` is set.